internal/inspect/detectors/plumber.go

```diff
@@ -3,38 +3,151 @@ package detectors
 // Copyright (C) 2023 by Posit Software, PBC.
 
 import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+
 	"github.com/posit-dev/publisher/internal/config"
 	"github.com/posit-dev/publisher/internal/contenttypes"
 	"github.com/posit-dev/publisher/internal/executor"
+	"github.com/posit-dev/publisher/internal/logging"
 	"github.com/posit-dev/publisher/internal/util"
 )
 
 type PlumberDetector struct {
 	inferenceHelper
+	log      logging.Logger
 	executor executor.Executor
 }
 
-func NewPlumberDetector() *PlumberDetector {
+type plumberServerMetadata struct {
+	Engine      string `yaml:"engine"`
+	Routes      any    `yaml:"routes"` // string when single file or []string when multiple files
+	Constructor string `yaml:"constructor"`
+}
+
+// rsconnect looks for these specific entrypointPath filenames and server files.
+var possiblePlumberEntrypoints = []string{"plumber.R", "entrypoint.R"}
```

Comment:
should these maybe also include lower case extensions `plumber.r`/`entrypoint.r`?